### PR TITLE
Added resolution for module 4 homework

### PR DESCRIPTION
**Pull request by anatoly-karpovich**
Added jasmine framework
Added ability to use jasmine instead of mocha
Added wrapper over loggers to get needed logger
Added 3 new tests (create - update - delete dashboard)
Added configuration files
Added ability to hide secret data in reports/logs

### CODE REVIEW

**Review by OlegGoncharuk — APPROVED** [review]
looks good to me

File: src/config/wdio.conf.ts [hunk 1]
```diff
@@ -125,7 +129,7 @@ export const config: Options.Testrunner = {
     //
     // Make sure you have the wdio adapter package for the specific framework installed
     // before running any tests.
-    framework: 'mocha',
+    // framework: 'mocha',
```
Comment (Vigidoriya) [inline src/config/wdio.conf.ts]: I believe this line could be removed

File: src/ui/pages/basePage.page.ts [hunk 2]
```diff
@@ -13,11 +24,7 @@ export class BasePage {
...
   }
 
-  async waitForElement(
-    selector: string,
-    reverse = false,
-    timeout = DEFAULT_TIMEOUT
-  ) {
+  async waitForElement(selector: string, reverse = false, timeout = DEFAULT_TIMEOUT) {
     const element = await this.findElement(selector);
     element.waitForDisplayed({
```
Comment (Vigidoriya) [inline src/ui/pages/basePage.page.ts]: I think 'await' is missed here

File: src/ui/pages/basePage.page.ts [hunk 3]
```diff
@@ -95,13 +102,35 @@ export class BasePage {
...
+
+  async checkNotificationWithText(text: string) {
+    return await browser.waitUntil(async () => {
+      const notifications = await this.findElementArray(this["Notification message"]());
+      let expectedNotification: WebdriverIO.Element | undefined;
+      for (const n of notifications) {
+        let actualText = await n.getText();
+        if (text === actualText) {
+          expectedNotification = n;
+          n.click();
```
Comment (Vigidoriya) [inline src/ui/pages/basePage.page.ts]: click is also async method

File: src/config/wdio.conf.ts (hunk 1 above)
Comment (anatoly-karpovich) [reply src/config/wdio.conf.ts]: Fixed

File: src/ui/pages/basePage.page.ts (hunk 2 above)
Comment (anatoly-karpovich) [reply src/ui/pages/basePage.page.ts]: Fixed

File: src/ui/pages/basePage.page.ts (hunk 3 above)
Comment (anatoly-karpovich) [reply src/ui/pages/basePage.page.ts]: Fixed

**Pull request merged**